src/parser/ParsedQuery.h

```diff
@@ -339,6 +338,9 @@ class ParsedQuery {
     // Sets the Selector to 'All' (*) only if the Selector is still undefined
     // Returned value maybe unused due to Syntax Check
     [[maybe_unused]] bool setsAsterisk() {
+      /*
+       * Needs (std::monostate_t) but unnecessary due to Syntax Check
+       */
       // if (!isAsterisk() && !isVariables()) {
         setAsterisk();
```

Comment:
Remove the comment (which doesn't mean anything here) and the outcommented code.
make the function `setAsterisk()` public, and use it, and remove the unused function `setsAsterisk`